Server/ReasnAPI/ReasnAPI/Controllers/EventsController.cs

```diff
@@ -1,171 +1,307 @@
     {
-        throw new NotImplementedException();
+        var interestedDto = eventService.GetEventParticipantsBySlugAndStatus(slug, ParticipantStatus.Interested).ToList();
+        var participatingDto = eventService.GetEventParticipantsBySlugAndStatus(slug, ParticipantStatus.Participating).ToList();
+
+        var participantsResponse = new ParticipantsResponse
+        {
+            Interested = interestedDto,
+            Participating = participatingDto
+        };
+
+        return Ok(participantsResponse);
     }
 
     [HttpGet]
     [Route("{slug}/comments")]
-    public IActionResult GetEventComments(string slug)
+    [ProducesResponseType<List<CommentDto>>(StatusCodes.Status200OK)]
+    public IActionResult GetEventComments([FromRoute] string slug)
     {
-        throw new NotImplementedException();
+        var commentsDto = eventService.GetEventCommentsBySlug(slug); 
+        return Ok(commentsDto);
     }
 
     [HttpPost]
     [Authorize]
     [Route("{slug}/comments")]
-    public IActionResult AddEventComment(string slug)
+    public IActionResult AddEventComment([FromRoute] string slug, [FromBody]CommentRequest commentRequest, [FromServices] IValidator<CommentDto> validator)
     {
-        throw new NotImplementedException();
+        var user = userService.GetCurrentUser();
+        var @event = eventService.GetEventBySlug(slug);
+        var commentDto = commentRequest.ToDtoFromRequest(user.Id, @event.Id);
+        validator.ValidateAndThrow(commentDto);
+        
+        eventService.AddEventComment(commentDto);
+        return Ok();
     }
 
-    [HttpPost]
+    [HttpGet]
     [Authorize(Roles = "Admin, Organizer")]
     [Route("{slug}/parameters")]
-    public IActionResult AddEventParameter(string slug)
+    [ProducesResponseType<List<ParameterDto>>(StatusCodes.Status200OK)]
+    public IActionResult GetEventsParameters([FromRoute] string slug)
     {
-        throw new NotImplementedException();
-    }
+        var @event = eventService.GetEventBySlug(slug);
+        var user = userService.GetCurrentUser();
+        if (user.Role != UserRole.Admin && @event.OrganizerId != user.Id)
+        {
+            Forbid();
+        }
 
-    [HttpDelete]
-    [Authorize(Roles = "Admin, Organizer")]
-    [Route("{slug}/parameters/{parameterId:int}")]
-    public IActionResult DeleteEventParameter(string slug, int parameterId)
-    {
-        throw new NotImplementedException();
-    }
-
-    [HttpPost]
-    [Authorize(Roles = "Admin, Organizer")]
-    [Route("{slug}/tags")]
-    public IActionResult AddEventTag(string slug)
-    {
-        throw new NotImplementedException();
-    }
-
-    [HttpDelete]
-    [Authorize(Roles = "Admin, Organizer")]
-    [Route("{slug}/tags/{tagId:int}")]
-    public IActionResult DeleteEventTag(string slug, int tagId)
-    {
-        throw new NotImplementedException();
+        var parameters = @event.Parameters.ToDtoList();
+        return Ok(parameters);
```

Comment:
Propably it wont be used since all users can get params/tags by just getting an event... This should imo be just /parameters and return ALL parameters (only keys, distinct, just to choose from on FE) from DB.

Reply:
changed